src/engine/gameObject.js

```diff
@@ -16,10 +16,83 @@ var GameObject = props => {
       self.frame = (self.frame + self.frameRate * dt) % self.frames.length;
     },
     draw: _ => {
-      paletteRenderer.draw(self.frames[~~self.frame], ~~self.x, ~~self.y, self.paletteIndex);
+      paletteRenderer.draw(self.frames[~~self.frame], ~~self.x, ~~self.y, self.paletteIndex, self.flipped);
     }
   };
   self._update = self.update;
   return self;
 };
 
+var partsConfig = {
+  heads: [
+    [0],
+    [6]
+  ],
+  torsos: [
+    [2],
+    [8]
+  ],
+  wheels: [
+    [4],
+    [8]
+  ],
+};
+
+var Robot = props => {
+  var self = GameObject(props);
+  self.update = dt => {
+    self.components.forEach(c => { 
+      c.x = self.x + c.dx;
+      c.y = self.y + c.dy;
+    });
+  }
+  self.setSprites = () => {
+    self.head1.frames = partsConfig.heads[self.head];
+    self.head2.frames = partsConfig.heads[self.head];
+    self.torso1.frames = partsConfig.torsos[self.torso];
+    self.torso2.frames = partsConfig.torsos[self.torso];
+    self.wheels1.frames = partsConfig.wheels[self.wheels];
+    self.wheels2.frames = partsConfig.wheels[self.wheels];
+  }
+  self.draw = noop; // We don't draw this gameObject, it's just a container
+  self.head1 = GameObject([0, 0, [1], i+3, self.paletteIndex]);
+  self.head1.dx = 0;
+  self.head1.dy = 0;
+  self.head2 = GameObject([0, 0, [1], i+3, self.paletteIndex]);
+  self.head2.flipped = true;
+  self.head2.dx = 16;
+  self.head2.dy = 0;
+  
+  self.torso1 = GameObject([0, 0, [1], i+3, self.paletteIndex]);
+  self.torso1.dx = 0;
+  self.torso1.dy = 16;
+  self.torso2 = GameObject([0, 0, [1], i+3, self.paletteIndex]);
+  self.torso2.flipped = true;
+  self.torso2.dx = 16;
+  self.torso2.dy = 16;
+
+  self.wheels1 = GameObject([0, 0, [1], i+3, self.paletteIndex]);
+  self.wheels1.dx = 0;
+  self.wheels1.dy = 32;
+  self.wheels2 = GameObject([0, 0, [1], i+3, self.paletteIndex]);
+  self.wheels2.flipped = true;
+  self.wheels2.dx = 16;
+  self.wheels2.dy = 32;
+
+  self.components = [];
+  self.components.push(self.head1);
+  self.components.push(self.head2);
+  self.components.push(self.torso1);
+  self.components.push(self.torso2);
+  self.components.push(self.wheels1);
+  self.components.push(self.wheels2);
```

Comment:
the declaration could be shorter, like: 
self.components = [self.head1, self.head2, self.torso1, self.torso2, self.wheels1, self.wheels2];